src/interrupt.c

```diff
@@ -31,9 +31,11 @@
  * in pairs.  Calling #enable_interrupts without first calling
  * #disable_interrupts is considered a violation of this assumption
  * and should be avoided.  Calling #disable_interrupts when interrupts
- * are already disabled will have no effect.  Calling #enable_interrupts
- * again to restore from a critical section will not enable interrupts
- * if interrupts were not enabled when calling #disable_interrupts.
+ * are already disabled will have no effect interrupts-wise
+ * (but should be paired with a #enable_interrupts regardless).
+ * Calling #enable_interrupts again to restore from a critical section
+ * will not enable interrupts if interrupts were not enabled when
+ * calling #disable_interrupts.
```

Comment:
Yeah it's weird but just
> Calling #disable_interrupts when interrupts are already disabled will have no effect
may be misleading in that it still pushes the "depth" down by one, so it does have a side effect

I haven't changed this paragraph, do you mean it could be trimmed down?
Maybe continue the sentence like,
>  \* , and in that case the paired #enable_interrupts will not enable interrupts either.